akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -759,6 +760,35 @@ private[akka] class ShardRegion(
           after.map(_.address).getOrElse(""))
       coordinator = None
       startRegistration()
```

Comment:
Leaving this as a final safety mechanism, even if it means we're effectively doing a pre-registration when the prior coordinator leaves/exits and then doing a full registration when we see the prior coordinator removed (depending on the ordering of the removals, we could do multiple rounds of full registration?)